### ISSUE TITLE
Version 1.16 - Eclipse 2024-03 MacOS Apple Silcon breaks WindowBuilder.

### ISSUE THREAD

**Issue by C4J** [eclipse-windowbuilder/windowbuilder#792]
I have just updated from 1.15 to 1.16.

The update process seems to take a lot longer than normal, specifically after it requests to restart eclipse there is a rather long delay while it restarts - maybe 2 mins in which time the CPU usage is over 400%.

When I attempt to view any GUI screens using WindowBuilder all I get is a white page with no features on it.

BEFORE UPDATE 1.15
<img width="1792" alt="before" src="https://github.com/eclipse-windowbuilder/windowbuilder/assets/20553038/9993676f-a17b-474e-a6c9-a551c7396afc">

AFTER UPDATE 1.16
<img width="1792" alt="after" src="https://github.com/eclipse-windowbuilder/windowbuilder/assets/20553038/b65ade04-0623-4a84-b8c5-ed5176afa802">

If I revert back to 1.15 then all works again.

I will try a clean install of Eclipse and see if that helps.

**Comment by C4J** [comment]
``I am installing into a clean fresh download of [Eclipse IDE for Enterprise Java and Web Developers](https://www.eclipse.org/downloads/packages/release/2024-03/r/eclipse-ide-enterprise-java-and-web-developers) AArch64 for MacOS.

I am then installing WindowBuilder Pro from the "latest" url.

I notice immediatly that the Eclipse updater is searching for alternative solutions and then tells me there are conflicts.

```
Cannot complete the install because of a conflicting dependency.
  Software being installed: WindowBuilder GroupLayout Support 1.16.0.202405081620 (org.eclipse.wb.layout.group.feature.feature.group 1.16.0.202405081620)
  Software currently installed: DTP SQL Schema Object Editor Feature 1.15.0.202311071249 (org.eclipse.datatools.sqldevtools.schemaobjecteditor.feature.feature.group 1.15.0.202311071249)
  Only one of the following can be installed at once: 
    GEF Classic Draw2d 3.16.0.202405290843 (org.eclipse.draw2d 3.16.0.202405290843)
    GEF Classic Draw2d 3.15.0.202402212051 (org.eclipse.draw2d 3.15.0.202402212051)
  Cannot satisfy dependency:
    From: DTP SQL Schema Object Editor Feature 1.15.0.202311071249 (org.eclipse.datatools.sqldevtools.schemaobjecteditor.feature.feature.group 1.15.0.202311071249)
    To: org.eclipse.equinox.p2.iu; org.eclipse.draw2d.feature.group 3.4.0
  Cannot satisfy dependency:
    From: GEF Classic Draw2d 3.19.0.202402212051 (org.eclipse.draw2d.feature.group 3.19.0.202402212051)
    To: org.eclipse.equinox.p2.iu; org.eclipse.draw2d [3.15.0.202402212051,3.15.0.202402212051]
  Cannot satisfy dependency:
    From: WindowBuilder GroupLayout Support 1.9.600.202405081620 (org.eclipse.wb.layout.group 1.9.600.202405081620)
    To: osgi.bundle; org.eclipse.draw2d 3.16.0
  Cannot satisfy dependency:
    From: WindowBuilder GroupLayout Support 1.16.0.202405081620 (org.eclipse.wb.layout.group.feature.feature.group 1.16.0.202405081620)
    To: org.eclipse.equinox.p2.iu; org.eclipse.wb.layout.group [1.9.600.202405081620,1.9.600.202405081620]
```

I decide to choose the option to "make my eclipse compatible" with the installed components which basically means it will remove a load of existing plugins.

See below.

<img width="1202" alt="fix" src="https://github.com/eclipse-windowbuilder/windowbuilder/assets/20553038/1b2483fe-2b83-4c0c-9653-7af3f0eb878f">

After selecting this Eclipse requests a restart and this still takes a lot longer than I would normally expect from a Mac Apple Silicon M3 with 96GB of RAM.

<img width="1006" alt="activity" src="https://github.com/eclipse-windowbuilder/windowbuilder/assets/20553038/cee87143-f299-4b74-9d27-7dfcbc6eebc6">

Once the IDE has restarted I am able to open the Java Form and see the layout as I would expect - however I'm not sure what the implications of the removal of plugings maybe elsewhere along the line.

Hope this helps.

Dave

**Comment by merks** [comment]
It's a very strange error.  The feature itself only set s a lower bound on the GEF feature:

![image](https://github.com/eclipse-windowbuilder/windowbuilder/assets/208716/5a11a409-385c-48d9-89c0-170dd1d4f88f)

So it should be happy with pretty much any version of that feature that's available today.

In the SimRel aggregation analyzer editor it shows that the requirement is resolved to this new version of GEF:

![image](https://github.com/eclipse-windowbuilder/windowbuilder/assets/208716/69b76666-c77f-446f-a84a-280e8874a0d6)

@C4J 

As a test, you might try installing/update from https://download.eclipse.org/staging/2024-06/ which contains all the latest stuff and is almost at the state of what will be released next week Wednesday for 2024-06...

**Comment by C4J** [comment]
Rather than a version number issue - is it due to there being 2 versions - I only mention this due to the comment in the log above :-   **Only one of the following can be installed at once**: 
 GEF Classic Draw2d 3.16.0.202405290843 (org.eclipse.draw2d 3.16.0.202405290843)
    GEF Classic Draw2d 3.15.0.202402212051 (org.eclipse.draw2d 3.15.0.202402212051)

**Comment by merks** [comment]
Yes, though it's just not clear what two things require specifically those two different version.  In SimRel I have this analysis view that shows for any installable unit which other units strictly require that version of the IU:

![image](https://github.com/eclipse-windowbuilder/windowbuilder/assets/208716/0df53823-66b5-4ce5-9d6a-21568baf6699)

So I can see there is only one version of org.eclipse.draw2d in the SimRel repo and all requirements on the things it provides are quite open in their range, none of them excluding any version < 4.0.0...

**Comment by C4J** [comment]
Ok - I just installed 2024-06 RC1 and then WindowBuilder 1.16 and it seemed to install without the aforementioned  issues which is good. I will stick with my current release until 2024-06 is officially released and then do clean installs.

Of course - it's totally unrelated by I wish Eclipse JavaScript Syntax validation would be a little more consistent. This version seems to be very upset about missing semi colons - I suspect the reported error is a red herring.

**Comment by ptziegler** [comment]
Let's briefly summarize what's happening:

- The Eclipse IDE for Enterprise Java and Web Developers comes with the GEF 3.19.0 feature installed.
- WindowBuilder 1.16.0 requires at least GEF 3.20.0

Because GEF 3.20.0 has not been released yet, the Eclipse IDE is unable to update GEF and therefore also unable to properly install WindowBuilder. The result is a broken installation where some plugins are part of 1.15.0 and some part of 1.16.0.

By using "make my eclipse compatible", you effectively removed the GEF feature (and everything depending on it) and instead installed the bundles that are part of the WindowBuilder update site (which only contains the GEF bundles, not the feature!).

So in short: Not all dependencies could be resolved because the 2024-06 is not out yet. As Ed mentioned, you can get around this problem by either using the 2024-06 staging repository (or directly the GEF milestone repository) or directly by using the RC1. In both cases, the missing 1.16.0 feature will be available.

**Comment by ptziegler** [comment]
I'm closing this as I don't believe that this is a bug but rather an unfortunate side-effect of being so close to the release.
